integration-cli/docker_cli_build_test.go

```diff
@@ -4526,85 +4527,85 @@ func TestBuildRenamedDockerfile(t *testing.T) {
 	}
 
 	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", "-f", "files/Dockerfile", "-t", "test2", ".")
-
 	if err != nil {
 		t.Fatal(err)
 	}
 	if !strings.Contains(out, "from files/Dockerfile") {
-		t.Fatalf("Should have used files/Dockerfile, output:%s", out)
+		t.Fatalf("test2 should have used files/Dockerfile, output:%s", out)
 	}
 
 	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", "--file=files/dFile", "-t", "test3", ".")
-
 	if err != nil {
 		t.Fatal(err)
 	}
 	if !strings.Contains(out, "from files/dFile") {
-		t.Fatalf("Should have used files/dFile, output:%s", out)
+		t.Fatalf("test3 should have used files/dFile, output:%s", out)
 	}
 
 	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", "--file=dFile", "-t", "test4", ".")
-
 	if err != nil {
 		t.Fatal(err)
 	}
 	if !strings.Contains(out, "from dFile") {
-		t.Fatalf("Should have used dFile, output:%s", out)
+		t.Fatalf("test4 should have used dFile, output:%s", out)
 	}
 
 	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", "--file=/etc/passwd", "-t", "test5", ".")
-
 	if err == nil {
-		t.Fatalf("Was supposed to fail to find passwd")
+		t.Fatalf("test5 was supposed to fail to find passwd")
 	}
-
 	if !strings.Contains(out, "The Dockerfile (/etc/passwd) must be within the build context (.)") {
-		t.Fatalf("Wrong error message for passwd:%v", out)
+		t.Fatalf("test5 - wrong error message for passwd:%v", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test5", "..")
-
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test6", "..")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("test6 failed: %s", err)
 	}
-
 	if !strings.Contains(out, "from Dockerfile") {
-		t.Fatalf("Should have used root Dockerfile, output:%s", out)
+		t.Fatalf("test6 should have used root Dockerfile, output:%s", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", ctx.Dir+"/files/Dockerfile", "-t", "test6", "..")
-
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", ctx.Dir+"/files/Dockerfile", "-t", "test7", "..")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("test7 failed: %s", err)
 	}
-
 	if !strings.Contains(out, "from files/Dockerfile") {
-		t.Fatalf("Should have used files Dockerfile - 2, output:%s", out)
+		t.Fatalf("test7 should have used files Dockerfile, output:%s", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test7", ".")
-
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test8", ".")
 	if err == nil || !strings.Contains(out, "must be within the build context") {
-		t.Fatalf("Should have failed with Dockerfile out of context")
+		t.Fatalf("test8 should have failed with Dockerfile out of context: %s", err)
 	}
 
-	out, _, err = dockerCmdInDir(t, "/tmp", "build", "-t", "test6", ctx.Dir)
+	out, _, err = dockerCmdInDir(t, "/tmp", "build", "-t", "test9", ctx.Dir)
```

Comment:
Actually, just using os.TempDir() in place of "/tmp" is good enough. I'm not creating files in there, I'm just cd-ing to that dir so I can test running "docker build" from outside of the build context.  No need to create a new tempDir.